pkg/ingester/recovery.go

```diff
@@ -210,7 +210,7 @@ func (r *ingesterRecoverer) Close() {
 			s.unorderedWrites = isAllowed
 
 			if !isAllowed && old {
-				err := s.chunks[len(s.chunks)-1].chunk.ConvertHead(headBlockType(isAllowed))
+				err := s.chunks[len(s.chunks)-1].chunk.ConvertHead(headBlockType(s.chunkFormat, isAllowed))
```

Comment:
this code actually doesn't make sense to me because `isAllowed` here is always going to be false. Anyways, not going to block the PR on this.